server/agents/agent.py

```diff
@@ -168,7 +200,10 @@ async def _run_async_stream_unstructured(
         if user_input:
             self.history.initialize_turn()
             self.current_user_input = user_input
-            self.history.add_message("user", user_input)
+            # Only add to history if we don't have multimodal messages
+            # (multimodal messages bypass history and are used directly)
+            if not self._multimodal_messages:
```

Comment:
I believe atomic agents had support for multimodal messages in history and we can port that over to the new agent implementation soon